### PR TITLE
More gracefully handle MediaCloud API error

### PR DESCRIPTION
**Pull request by scttnlsn**

### CODE REVIEW

File: app/controllers/queries_controller.rb [hunk 1]
```diff
@@ -22,30 +22,34 @@ def create
...
-        Queries::DownloadFullData.(query: query)
-        send_data(presenter.render_csv, type: 'text/csv', filename: csv_file_name, disposition: :attachment.to_s)
+        begin
+          Queries::DownloadFullData.(query: query)
+          send_data(presenter.render_csv, type: 'text/csv', filename: csv_file_name, disposition: :attachment.to_s)
+        rescue MediaCloudQuery::APIError
+          puts err
+          Rollbar.error(err)
+          flash[:error] = 'Media Cloud not responding'
+          redirect_to project_path(project)
```
Comment (scttnlsn) [inline app/controllers/queries_controller.rb]: This is not ideal but I'm not sure what else we can do at this point.  We could alternatively leave an error message in the CSV output (also not ideal) or skip downloading the full dataset.  @sapraaman Any suggestions?

File: app/controllers/queries_controller.rb [hunk 2]
```diff
@@ -22,30 +22,34 @@ def create
...
     redirect_to project_query_path(project, @query)
   rescue ActiveRecord::RecordInvalid
-    render :new
-    respond_to(&:js)
+    render_error
   rescue MediaCloud::ParseError
-    @query.errors.add(:media_cloud_url, '^Unable to parse URL')
-    render :new
-    respond_to(&:js)
+    render_error('^Unable to parse URL')
```
Comment (sapraaman) [inline app/controllers/queries_controller.rb]: What does '^' do @scttnlsn ?

File: app/controllers/queries_controller.rb (hunk 1 above)
Comment (sapraaman) [reply app/controllers/queries_controller.rb]: No, I like this

File: app/controllers/queries_controller.rb [hunk 3]
```diff
@@ -22,30 +22,34 @@ def create
...
-    @query.errors.add(:media_cloud_url, '^Query timeout. Please try using Media Cloud directly')
-    render :new
-    respond_to(&:js)
+    render_error('^Query timeout. Please try using Media Cloud directly')
+  rescue MediaCloudQuery::APIError
+    puts err
+    Rollbar.error(err)
+    render_error('^Media Cloud not responding')
   rescue Exception => err
     puts err
```
Comment (sapraaman) [inline app/controllers/queries_controller.rb]: Shoudl we remove this ?

File: app/controllers/queries_controller.rb [hunk 4]
```diff
@@ -22,30 +22,34 @@ def create
...
-    respond_to(&:js)
+    render_error('^Unable to parse URL')
   rescue Net::ReadTimeout
-    @query.errors.add(:media_cloud_url, '^Query timeout. Please try using Media Cloud directly')
-    render :new
-    respond_to(&:js)
+    render_error('^Query timeout. Please try using Media Cloud directly')
+  rescue MediaCloudQuery::APIError
+    puts err
+    Rollbar.error(err)
```
Comment (sapraaman) [inline app/controllers/queries_controller.rb]: Rollbar.error should be part of `render_error`

File: app/controllers/queries_controller.rb (hunk 2 above)
Comment (scttnlsn) [reply app/controllers/queries_controller.rb]: It's the flag for https://github.com/jeremydurham/custom-err-msg

File: app/controllers/queries_controller.rb (hunk 3 above)
Comment (scttnlsn) [reply app/controllers/queries_controller.rb]: I think we should continue writing exceptions to the log but perhaps `logger.error(err)` is more appropriate

File: app/controllers/queries_controller.rb (hunk 4 above)
Comment (scttnlsn) [reply app/controllers/queries_controller.rb]: But then we'd be logging `MediaCloud::ParseError` and `ActiveRecord::RecordInvalid` errors to Rollbar.  I don't think we want that.

**Pull request merged**